src/download/exchangeDownloader.ts

```diff
@@ -20,12 +21,40 @@ import {
 } from "./exchangeTypes";
 import { ramlToolLogger } from "../common/logger";
 import { extractFile } from "./exchangeDirectoryParser";
+import { retryOptions } from "./config";
 
-const DEFAULT_DOWNLOAD_FOLDER = "download";
+export const DEFAULT_DOWNLOAD_FOLDER = "download";
 const ANYPOINT_BASE_URI = "https://anypoint.mulesoft.com/exchange";
 const ANYPOINT_API_URI_V1 = `${ANYPOINT_BASE_URI}/api/v1`;
 const ANYPOINT_API_URI_V2 = `${ANYPOINT_BASE_URI}/api/v2`;
 
+/**
+ * Makes an HTTP call to the url with the options passed. If the calls due to
+ * a 5xx, 408, 420 or 429, it retries the call with the retry options passed
+ * in the options argument. If no retry options are passed, it uses the default
+ * options set in retryOptions of config.ts.
+ *
+ * @param url - Where the request should be made
+ * @param options - options to be passed when the call is made
+ *
+ * @returns a promise with the raw response returned by the server
+ */
+export async function runFetch(
+  url: string,
+  options: { [key: string]: any } = {}
+): Promise<any> {
+  options.retry = _.merge({}, retryOptions, options.retry);
+  options = _.merge(options, {
+    onRetry() {
+      ramlToolLogger.info(
```

Comment:
Is there a way to log the error before retry or does it log by default?

Reply:
I don't see anything like that available with make-fetch-happen